content/docs/rfcs/20/README.md

```diff
@@ -137,6 +137,47 @@ message PublicKeyMessage {
 This MUST be wrapped in a Waku Message version 0, with the Public Key Broadcast content topic.
 Finally, Bob SHOULD publish the message on Waku v2. 
 
+## Consideration for a non-interactive protocol
+
+Alice has to get Bob's public Key to send a message to Bob.
+Because an Ethereum Address is part of the hash of the public key's account,
+it is not enough in itself to deduce Bob's Public Key.
+
+This is why the protocol dictates that Bob MUST send his Public Key first,
+and Alice MUST receive it before she can send him a message.
+
+Moreover, nim-waku, the reference implementation of [13/WAKU2-STORE](/spec/13/),
+stores messages for a maximum period of 30 days.
+This means that Bob would need to broadcast his public key at least every 30 days to be reachable.
+
+Below we are reviewing possible solutions to mitigate this "sign up" step.
+
+### Retrieve the public key from the blockchain
+
+If Bob has signed at least one transaction with his account then his Public Key can be extracted from the transaction's ECDSA signature.
+The drawback of this method is that standard Web3 Wallet APIs does not allow Alice to specifically retrieve all/any transaction sent by Bob.
+
+Alice would instead need to use the `eth.getBlock` API to retrieve Ethereum blocks one by one.
+For each block, she would need to check the `from` value of each transaction until she finds a transaction sent by Bob.
```

Comment:
I believe The Graph can be used here? cc @bgits

Reply:
We could index with The Graph to create a more efficient query.

Reply:
If I understand correctly this would mean creating a subgraph that index public keys against Ethereum addresses, extracting them from transaction?